agent/engine/docker_task_engine.go

```diff
@@ -248,9 +251,71 @@ func (engine *DockerTaskEngine) Init(ctx context.Context) error {
 	// Now catch up and start processing new events per normal
 	go engine.handleDockerEvents(derivedCtx)
 	engine.initialized = true
+	go engine.startPeriodicExecAgentsMonitoring(derivedCtx, monitorExecAgentsInterval)
 	return nil
 }
 
+func (engine *DockerTaskEngine) startPeriodicExecAgentsMonitoring(ctx context.Context, monitorInterval time.Duration) {
+	engine.monitorExecAgentsTicker = time.NewTicker(monitorInterval)
+	for {
+		select {
+		case <-engine.monitorExecAgentsTicker.C:
+			go engine.monitorExecAgentProcesses(ctx)
+		case <-ctx.Done():
+			engine.monitorExecAgentsTicker.Stop()
+			return
+		}
+	}
+}
+
+func (engine *DockerTaskEngine) monitorExecAgentProcesses(ctx context.Context) {
+	// TODO: add jitter between containers to not overload docker with top calls
+	for _, mTask := range engine.managedTasks {
+		task := mTask.Task
+		if !task.ExecCommandAgentEnabled {
+			return
+		}
+		for _, c := range task.Containers {
+			go engine.monitorExecAgentRunning(ctx, task.Arn, c)
+		}
+	}
+}
+
+func (engine *DockerTaskEngine) monitorExecAgentRunning(ctx context.Context,
+	tArn string, c *apicontainer.Container) {
+	if c.ExecCommandAgentMetadata == nil {
+		return
+	}
+	pID := c.GetExecCommandAgentMetadata().PID
+	cName := c.Name
+	cID := c.RuntimeID
+	if cID == "" || pID == "" {
+		return
+	}
+	resp, err := engine.client.TopContainer(ctx, cID, dockerclient.TopContainerTimeout, pID)
```

Comment:
the docker exec inspect output should also give us a clue about the status of our running agent -- 
for a failure I wonder if we can't also ask docker exec inspect what happened (we get the process exit code from that output)
```
    "ID": "f086903815fbbd127ec816eb154d772696088cc53ba22bbef0098bb8cc44849a",
    "Running": false,
    "ExitCode": 2,
```

Reply:
thats a good point and I considered it earlier, but its adding yet another docker API call, which is why I was reluctant. I will add a `TODO` to reconsider this call and checking based on exit codes.